runtime/compiler/ilgen/Walker.cpp

```diff
@@ -5478,17 +5499,15 @@ break
       push(resultNode);
       }
 
-   // callNodeTreeTop may be null if this call should not be placed in the trees
-   if (callNodeTreeTop
-       && comp()->getOption(TR_EnableOSR)
-       && !comp()->isPeekingMethod()
-       && comp()->isOSRTransitionTarget(TR::postExecutionOSR)
-       && comp()->isPotentialOSRPoint(callNodeTreeTop->getNode())
-       && !_methodSymbol->cannotAttemptOSRAt(callNode->getByteCodeInfo(), NULL, comp())
-       && !_cannotAttemptOSR)
+   if (needOSRBookkeeping)
       {
       saveStack(-1, !comp()->pendingPushLivenessDuringIlgen());
-      stashPendingPushLivenessForOSR(comp()->getOSRInductionOffset(callNode));
+      stashPendingPushLivenessForOSR(osrInductionOffset);
+      if (comp()->supportsInduceOSR())
```

Comment:
I think you might want getOSRMode() == TR::voluntaryOSRMode to be sure we don't try and do this for involuntary OSR.